src/mrack/transformers/beaker.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Beaker transformer module."""
+import re
+
+from mrack.transformers.transformer import Transformer
+from mrack.utils import get_config_value, print_obj
+
+CONFIG_KEY = "beaker"
+
+
+class BeakerTransformer(Transformer):
+    """Beaker transformer."""
+
+    _config_key = CONFIG_KEY
+    _required_config_attrs = ["distros", "keypair", "reserve_duration", "max_attempts"]
+
+    async def init_provider(self):
+        """Initialize associate provider."""
+        await self._provider.init(self.config)
```

Comment:
Note for the future: I wonder how tightly we are coupling XProvider with XTransformer. The `init` method for `AWSProvider` and `OpenStackProvider` is `async def init(self, image_names=None)`.

It's not a blocker here, based on my understanding on how mrack is currently working, but perhaps something we need to consider eventually.